backend/src/middleware/auth/checks/learning-content-auth-checks.ts

```diff
@@ -0,0 +1,20 @@
+import { authorize } from './auth-checks';
+import { AuthenticationInfo } from '../authentication-info';
+import { AuthenticatedRequest } from '../authenticated-request';
+
+/**
+ * Only allows requests whose learning path personalization query parameters ('forGroup' / 'assignmentNo' / 'classId')
+ * are
+ * - either not set
+ * - or set to a group the user is in,
+ * - or set to anything if the user is a teacher.
+ */
+export const onlyAllowPersonalizationForOwnGroup = authorize(async (auth: AuthenticationInfo, req: AuthenticatedRequest) => {
+    const { forGroup, assignmentNo, classId } = req.params;
+    if (auth.accountType === 'student' && forGroup && assignmentNo && classId) {
```

Comment:
We zouden beter ergens een enum maken voor de mogelijke accounttypes.
```suggestion
    if (auth.accountType === AccountType.Student && forGroup && assignmentNo && classId) {
```

Reply:
Aangepast in fc0d3b5c84f1dd898a25788c92be84680a9683a5